### PR TITLE
[MA-39] - Write to Keyword Update tables

### PR DESCRIPTION
**Pull request by balkrishnada**
Submitting the WIP PR from today.

- [x] Repo complete

- [x] Fix test failures - The current test is failing and the SELECT query does not fetch the inserted record.

- [x] Add test for Geo keyword update

- [x] Successful build

### CODE REVIEW

File: src/main/java/com/reachlocal/media/automation/keywordupdate/KeywordUpdate.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package com.reachlocal.media.automation.keywordupdate;
+
+public class KeywordUpdate {
+
+    private Long publisherCampaignId;
+    private Long keywordId;
+    private KeywordUpdateType keywordUpdateType;
+    private KeywordMatchType keywordMatchType;
+    private Long adGroupId;
```
Comment (balkrishnada) [inline src/main/java/com/reachlocal/media/automation/keywordupdate/KeywordUpdate.java]: Figuring out whether we need `adGroupId` to be set in the update table.

**Comment by balkrishnada** [comment]
Rebased.

Successful build: https://jenkins.media.dev.usa.reachlocalservices.com/view/media-automation/job/media-automation-build/57/

File: src/main/resources/sql/keywordupdate/keywordupdate.properties [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+VALUES (\
+  :wpcId,\
+  :keywordId,\
+  :keywordUpdateType,\
+  :keywordMatchType,\
+  :adGroupId)\
+  ON DUPLICATE KEY UPDATE \
+  KeywordMatchType_idKeywordMatchType_FK=VALUES(KeywordMatchType_idKeywordMatchType_FK), \
+  AdGroup_idAdGroup_FK=VALUES(AdGroup_idAdGroup_FK)
+KEYWORD_UPDATE.INSERT_GEO_KEYWORD_UPDATE=\
```
Comment (jellyvora-reachlocal) [inline src/main/resources/sql/keywordupdate/keywordupdate.properties]: add a line space above

File: src/main/java/com/reachlocal/media/automation/keywordupdate/KeywordUpdateRepository.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    public void createGeoKeywordUpdate(GeoKeywordUpdate geoKeywordUpdate) {
+        opJdbcTemplate.update(geoKeywordUpdateQuery, geoKeywordModelToRow(geoKeywordUpdate));
+    }
+
+    private MapSqlParameterSource keywordModelToRow(KeywordUpdate model) {
+
+        MapSqlParameterSource parameters = new MapSqlParameterSource();
+
```
Comment (jellyvora-reachlocal) [inline src/main/java/com/reachlocal/media/automation/keywordupdate/KeywordUpdateRepository.java]: dont need space here

File: src/main/java/com/reachlocal/media/automation/keywordupdate/KeywordUpdateRepository.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    public void createKeywordUpdate(KeywordUpdate keywordUpdate) {
+        opJdbcTemplate.update(keywordUpdateQuery, keywordModelToRow(keywordUpdate));
+    }
+
+    public void createGeoKeywordUpdate(GeoKeywordUpdate geoKeywordUpdate) {
+        opJdbcTemplate.update(geoKeywordUpdateQuery, geoKeywordModelToRow(geoKeywordUpdate));
+    }
+
+    private MapSqlParameterSource keywordModelToRow(KeywordUpdate model) {
+
```
Comment (jellyvora-reachlocal) [inline src/main/java/com/reachlocal/media/automation/keywordupdate/KeywordUpdateRepository.java]: dont need space here. use blank lines between codeblocks for logic separation.

File: src/main/java/com/reachlocal/media/automation/keywordupdate/KeywordUpdateRepository.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        parameters.addValue("keywordMatchType",
+                            model.getKeywordMatchType()
+                                 .getMatchTypeId());
+        parameters.addValue("adGroupId", model.getAdGroupId());
+
+        return parameters;
+    }
+
+    private MapSqlParameterSource geoKeywordModelToRow(GeoKeywordUpdate model) {
+
```
Comment (jellyvora-reachlocal) [inline src/main/java/com/reachlocal/media/automation/keywordupdate/KeywordUpdateRepository.java]: dont need space here

File: src/main/java/com/reachlocal/media/automation/keywordupdate/KeywordUpdateRepository.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        MapSqlParameterSource parameters = keywordModelToRow(model);
+
```
Comment (jellyvora-reachlocal) [inline src/main/java/com/reachlocal/media/automation/keywordupdate/KeywordUpdateRepository.java]: dont need space here

File: src/main/java/com/reachlocal/media/automation/keywordupdate/KeywordUpdateRepository.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        parameters.addValue("geoKeywordId", model.getGeoKeywordId());
+
```
Comment (jellyvora-reachlocal) [inline src/main/java/com/reachlocal/media/automation/keywordupdate/KeywordUpdateRepository.java]: dont need space here

File: src/main/java/com/reachlocal/media/automation/keywordupdate/KeywordUpdateRepository.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        parameters.addValue("wpcId", model.getPublisherCampaignId());
+        parameters.addValue("keywordId", model.getKeywordId());
+        parameters.addValue("keywordUpdateType",
+                            model.getKeywordUpdateType()
+                                 .getUpdateTypeId());
+        parameters.addValue("keywordMatchType",
+                            model.getKeywordMatchType()
+                                 .getMatchTypeId());
+        parameters.addValue("adGroupId", model.getAdGroupId());
+
```
Comment (jellyvora-reachlocal) [inline src/main/java/com/reachlocal/media/automation/keywordupdate/KeywordUpdateRepository.java]: dont need space here

File: src/main/java/com/reachlocal/media/automation/keywordupdate/KeywordMatchType.java [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    public static KeywordMatchType of(Integer id) {
+        for (KeywordMatchType matchType : KeywordMatchType.values()) {
+            if (matchType.getMatchTypeId()
+                         .equals(id)) {
+                return matchType;
+            }
+        }
+        throw new IllegalArgumentException("Positive Keyword Match Type not found");
```
Comment (jellyvora-reachlocal) [inline src/main/java/com/reachlocal/media/automation/keywordupdate/KeywordMatchType.java]: `throw new IllegalArgumentException("Keyword Match Type not found");

**Pull request merged**